tests/support.py

```diff
@@ -1,24 +1,85 @@
+""" Support functions for tests and benchmarking. """
+
+import importlib.util
+import glob
+import os
 import sys
-from pathlib import Path
+
 import hpy.universal
 
-def import_ujson_universal():
+
+def import_ujson_hpy(abi="default"):
+    """ Either load the default ujson_hpy, or specifically the ujson_hpy
+        universal module if the UJSON_UNIVERSAL environment variable is
+        set to a non-empty value.
     """
-    Find and import ujson.hpy.so.
+    abi = os.environ.get("HPY_ABI", abi)
+    if abi == "default":
+        import ujson_hpy
+        mod = ujson_hpy
+    elif abi == "cpython":
+        mod = load_cpython_module(
+            "ujson_hpy",
+            find_so_filename("ujson_hpy.cpython-*.so"))
+    elif abi == "universal":
+        mod = load_universal_module(
+            "ujson_hpy",
+            find_so_filename("ujson_hpy.hpy.so"),
+            debug=False)
+    elif abi == "debug":
+        mod = load_universal_module(
+            "ujson_hpy",
+            find_so_filename("ujson_hpy.hpy.so"),
+            debug=True)
+    else:
+        raise ValueError("Unknown HPY_ABI {!r}".format(abi))
+    return mod
 
-    Eventually this logic should be provided directly by hpy_universal,
-    but in the meantime we just do it manually.
+
+def find_so_filename(pattern):
+    """ Search for an extension module .so file that matches the specified
+        pattern.
     """
-    # find ujson_hpy.hpy.so in sys.path
     for p in sys.path:
-        sopath = Path(p) / 'ujson_hpy.hpy.so'
-        if sopath.exists():
-            break
-    else:
-        raise ImportError('Cannot find ujson.hpy.so')
-    #
-    # load it through hpy_universal
-    class Spec:
-        name = 'ujson_hpy'
-        origin = str(sopath)
-    return hpy.universal.load_from_spec(Spec)
+        paths = glob.glob(os.path.join(p, pattern))
+        if not paths:
+            continue
+        if len(paths) > 1:
+            raise RuntimeError(
+                "Found multiple possible .so files for pattern {!r} in"
+                " sys.path entry {!r}".format(pattern, p))
+        return paths[0]
+    raise RuntimeError(
+        "Could not find .so file for pattern {!r} in sys.path".format(pattern))
+
+
+def load_universal_module(name, so_filename, debug):
+    """ Load a universal ABI module .so file.
+
+        Copied from hpy/tests/support.py.
+    """
+    assert name not in sys.modules
+    mod = hpy.universal.load(name, so_filename, debug=debug)
+    mod.__file__ = so_filename
+    return mod
+
+
+def load_cpython_module(name, so_filename):
+    """ Load a cpython ABI module .so file.
+
+        Copied from hpy/tests/support.py.
+    """
```

Comment:
this is good, but it's probably a good hint that we should put this code inside `hpy.importing` or something like that

Reply:
Agreed.

Reply:
absoultely